### DISCUSSION TITLE
"no bottle available!" error when following the Formula Cookbook guide

### DISCUSSION THREAD

**Discussion by Southpaw1496** [Homebrew/discussions#3271] [Writing Formulae/Casks]
I'm trying to make a formula for the [Minecraft Ferium CLI](https://github.com/theRookieCoder/ferium/), and I've been following the [Homebrew Formula Cookbook](https://docs.brew.sh/Formula-Cookbook). It said to start by running `brew create` with a link to the source Tarball, and I'd read elsewhere in the docs that there are templates for various types of software, so since this is a Rust program I ran `brew create --rust https://github.com/theRookieCoder/ferium/archive/refs/tags/v3.28.7.tar.gz`. 

After filling out the `homepage` and `license` fields as instructed, the next step was to "Check the Build System", so I ran `brew install --interactive ferium`. I thought this was a little odd because I know Homebrew generally works with bottles, and I hadn't set any up yet, but I assumed it would come later in the article. However, when I ran the command, I received the error:
```
Error: ferium: no bottle available!
You can try to install from source with:
  brew install --build-from-source ferium
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```
The cookbook seems to assume this step will be successful, so I'm not sure how to proceed.

Here is my `ferium.rb` file:
```ruby
# Documentation: https://docs.brew.sh/Formula-Cookbook
#                https://rubydoc.brew.sh/Formula
# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
class Ferium < Formula
  desc "Ferium is a CLI Minecraft mod manager for mods from Modrinth, CurseForge, and Github Releases"
  homepage "https://github.com/theRookieCoder/ferium/"
  url "https://github.com/theRookieCoder/ferium/archive/refs/tags/v3.28.7.tar.gz"
  sha256 "ddedea0fcad3765da4faec648d8b21be469fe335351a40a7f00237706d178eb0"
  license "MPL-2.0"

  depends_on "rust" => :build

  def install
    # ENV.deparallelize  # if your formula fails when building in parallel
    system "cargo", "install", *std_cargo_args
  end

  test do
    # `test do` will create, run in and delete a temporary directory.
    #
    # This test will fail and we won't accept that! For Homebrew/homebrew-core
    # this will need to be a test that verifies the functionality of the
    # software. Run the test with `brew test ferium`. Options passed
    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
    #
    # The installed folder is not in the path, so use the entire path to any
    # executables being tested: `system "#{bin}/program", "do", "something"`.
    system "false"
  end
end
```

And the output of `brew -v`
```
Homebrew 3.4.11-34-gc20b3ab
Homebrew/homebrew-core (git revision 04c73a4f47a; last commit 2022-05-15)
Homebrew/homebrew-cask (git revision 2593a796f4; last commit 2022-05-15)
```

**Comment by SMillerDev** [comment]
Without bottles you need to pass`--build-from-source`. The bottles will be available after CI merges the pull request

**Reply by Southpaw1496** [reply]
But then why isn't that in the cookbook?

**Reply by SMillerDev** [reply]
Likely because that particular comment is outdated and nobody noticed it yet.
